qcodes/utils/metadata.py

```diff
@@ -1,6 +1,13 @@
+import json
+
 from typing import Sequence
 from .helpers import deep_update
+from functools import partial
 
+from typing import Dict, Tuple, Any, NamedTuple, Generic, TypeVar
+T = TypeVar('T')
+Snapshot = Dict[str, T] # TODO: represent known keys in typing
```

Comment:
that would be exciting! then this type definition can go to `qcodes/station.py` and be reused everywhere :)

Reply:
What I need to do is figure out how to make a Python type hint that accepts only `dict`s which have at least a given set of keys.

Reply:
as discussed offline, let's keep it simple for now until qcodes drops support for python 3.6 and we can make the heavy use of data classes of python 3.7. but please remove the "todo" :)

Reply:
Done!